scripts/provisioning.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+################################################################################
+################################################################################
+#
+# This script calls the needed DebOps plays/roles in order to provision a new cloud VM,
+# then prepares the VM to be ansible managed and runs additional Ansible Playbooks in the correct order.
+#
+# Usage:
+#     ./provisioning.sh -l <host-or-group-to-provision>
+#
+#     You can also pass ansible parameter to DebOps
+#     ./provisioning.sh -l <host-or-group-to-provision> -t role::pki --skip-tags role::owncloud --user root -vvv
+#
+################################################################################
+
+set -o nounset -o pipefail -o errexit
+
+# Check if we are in the correct root folder of the repo, and change upwards if needed
+# otherwise debops won't work
+if [[ ! -f .debops.cfg && -f ../.debops.cfg ]]; then
+    echo "Running $0 from root of repo."
+    pushd .. > /dev/null
+fi
+
+# Run the bootstrap DebOps playbook to make the VM a DebOps/Ansible managed host.
+debops --become $@ bootstrap-ldap || debops --become --user root $@ bootstrap-ldap
```

Comment:
Since this script seems to be meant to provision standalone hosts, you most likely meant to use the `bootstrap` playbook here, not `bootstrap-ldap` playbook.

I'm not really sure about the `||` usage to handle connections either to the user account, or the `root` account. This will run Ansible twice in case of an error on the first command, which might be confusing. Besides, with `-o pipefail` the script will stop when the first command returns with an error, correct?

Reply:
During initial provisioning there is only a root account so it runs the second debops command. After logging an error, but not failing the script. The `||`  (OR) keeps the script ongoing and works for one of/or both cases.

The pipefail has to do with `|` (pipe) not `||` (OR). The pipefail makes debugging easier, because wrongful output is not piped in the next command.

Reply:
AFAIK if the bootstrap(-ldap) plays have the `become: yes`  flag, there is no need to execute the bootstraping with `--become`. I can make a PR for this.

Reply:
The `bootstrap` and `bootstrap-ldap` playbooks already have the `become` keyword, so you don't need to specifically make PR for that. However I still think that here you want to use the `bootstrap` playbook, and not `bootstrap-ldap`, since you have a separate script for that.

Reply:
Fixed typo.